File: 0013.RomanToInteger/RomanToInteger.py
class Solution:
    def romanToInt(self, s: str) -> int:
        romanmap = {'I':1, 'V':5, 'X':10, 'L':50, 'C':100, 'D':500, 'M':1000}
        slen = len(s)
        i,ret = 0,0
        
        while i < slen - 1:
            if romanmap[s[i]] < romanmap[s[i + 1]]:
                ret += romanmap[s[i + 1]] - romanmap[s[i]]
                i += 2
            else:
                ret += romanmap[s[i]]
                i += 1
        if i == slen - 1:
            ret += romanmap[s[-1]]
        return ret

File: 0013.RomanToInteger/RomanToInteger.c


int romanToInt(char * s){
    int map[26], ret = 0;
    map['I' - 'A'] = 1;
    map['V' - 'A'] = 5;
    map['X' - 'A'] = 10;
    map['L' - 'A'] = 50;
    map['C' - 'A'] = 100;
    map['D' - 'A'] = 500;
    map['M' - 'A'] = 1000;
    
    while (*s) {
        if (!s[1]) {
            ret += map[s[0] - 'A'];
            break;
        }
        
        if (map[s[0] - 'A'] < map[s[1] - 'A']) {
            ret += map[s[1] - 'A'] - map[s[0] - 'A'];
            s += 2;
        } else {
            ret += map[s[0] - 'A'];
            s += 1;
        }
    }
    
    return ret;
}

